### DISCUSSION TITLE
Conditional SSG

### DISCUSSION THREAD

**Discussion by tjdam** [vercel/next.js#12470] [Help] [answered]
Is it possible to conditionally access SSG content?
Like, if I have a portion of my website that is always the same for not logged users but when a user is logged, the same url turns into a particular version of it for this particular user.

**Comment by brianlovin** [comment, marked as answer]
SSG constructs pages at build time, so it wouldn't know about any kind of user data or authentication. So as far as I know, no, this isn't possible to SSG a user-specific bit of content (at least, not if it requires authentication). 

It might be worth looking into how the Vercel folks have implemented SSG for their dashboard views: basically each internal page can load an SSG'd "skeleton" of the page. Once it mounts on the client, user access is validated and either populates the content of the skeleton, or redirects out to a login page.

**Reply by tjdam** [reply]
That sounds like exactly what I'm looking for! Is that code openly available?!

**Reply by brianlovin** [reply]
I'm not sure, there's probably a snippet floating around out there. I remember hearing Tim describe how it worked in http://www.fullstackradio.com/137, if you are up for a listen.

It's probably something like:

```js
import useSWR from 'swr'

function Page() {
  const { data, error } = useSWR('/api/user', fetcher)

  if (error) return <ErrorView />
  if (!data) return <Skeleton />
  if (data && !data.user) return <Login /> 
  return <Dashboard user={data.user} />
}
```

**Reply by timneutkens** [reply]
I went into it in-depth here: https://github.com/zeit/next.js/discussions/10724#discussioncomment-720
